runestone/common/runestonedirective.py

```diff
@@ -261,10 +264,33 @@ def __init__(self, *args, **kwargs):
         else:
             self.int_points = 1
 
-        self.options['optclass'] = self.options.get('class', "")
+        self.options["optclass"] = self.options.get("class", "")
 
         self.explain_text = []
 
+    # Check for a `valid HTML5 divid <https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute>`_.
+    def validate_divid(self, divid):
+        if (
+            # Per the spec, a divid must not contain `whitespace <https://infra.spec.whatwg.org/#ascii-whitespace>`_ (see also `Python string escape sequences <https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals>`_).
```

Comment:
Additionally, for XML compatibility divids should not begin with a numeric character.

Reply:
We have a handful of exercises in existing books that are just something like 3_1, 3_2 -- these will need to be migrated to a new name.

Reply:
I'm adding this to the code...which is also producing some pain for me -- I used random alphanumeric strings as divids, so I have some fixup to do on my book as well.